src/Symfony/Component/Validator/Tests/LegacyValidatorTest.php

```diff
@@ -39,4 +42,48 @@ public function testValidateValueRejectsValid()
     {
         $this->validator->validateValue(new Entity(), new Valid());
     }
+
+    public function testValidateMultipleGroupsForCollectionConstraint()
+    {
+        $entity = new Entity();
+        $entity->firstName = array('baz' => 2);
+
+        $this->metadata->addPropertyConstraint('firstName', new Collection(
+            array(
+                'fields' => array(
+                    'baz' => array(
+                        new Range(array('min' => 3, 'minMessage' => 'Group foo', 'groups' => 'foo')),
+                        new Range(array('min' => 5, 'minMessage' => 'Group bar', 'groups' => 'bar')),
+                    ),
+                ),
+            )
+        ));
```

Comment:
If there is no group set on the collection contraint, group on entry should never be validated (since it won't ever be a subset).